app/containers/Dashboard.jsx

```diff
@@ -1,10 +1,32 @@
-import React from 'react';
-
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 /*
  * Note: This is kept as a container-level component,
  *  i.e. We should keep this as the container that does the data-fetching
  *  and dispatching of actions if you decide to have any sub-components.
  */
-const Dashboard = () => <div>Welcome to the Dasboard. Stay tuned...</div>;
+class Dashboard extends Component {
+  componentWillMount() {
+    const { user: { authenticated }, history } = this.props;
```

Comment:
Is it possible to couple this with route config as in rr3? Or is this the new way of protecting routes? If so, maybe this should be in it's own component (or HoC)? Usage would be like

```javascript
<Protected redirect={'/login'}>
    <MyComponent/>
</Protected>
```

Reply:
Here is an example of a HOC doing this:

```javascript
import {Component} from 'react'
import PropTypes from 'prop-types'
import {connect} from 'react-redux'

const asProtected = (WrappedComponent, selectData) => {
    class Protected extends Component {
        static propTypes = {
            user: PropTypes.object,
            history: PropTypes.object
        }
        
        componentWillMount() {
            const {user: {authenticated}, history} = this.props
            
            if (!authenticated) {
                history.replace('/login')
            }
        }

        render() {
            return <WrappedComponent {...this.props}/>
        }
    }
    
    const mapStateToProps = ({user}) => ({
        user
    })
    
    return connect(
        mapStateToProps
    )(Protected)
}

export default asProtected
```

Usage:
```javascript
import asProtected from 'hoc/asProtected'

const MyComp = () => (
    <div>
        Only for logged in users!
    </div>
)

export default asProtected(MyComp)
```

Reply:
The same could probably be done for redirections.